eng/packages/http-client-csharp-mgmt/emitter/test/resolve-arm-resources-validation.test.ts

```diff
@@ -0,0 +1,1276 @@
+}
+
+interface Operations extends Azure.ResourceManager.Operations {}
+
+@armResourceOperations
+interface Employees {
+  get is ArmResourceRead<Employee>;
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<Employee>;
+}
+
+@armResourceOperations
+interface CurrentEmployees {
+  get is ArmResourceRead<CurrentEmployee>;
+  createOrUpdate is ArmResourceCreateOrReplaceAsync<CurrentEmployee>;
+}
+`;
+
+  /**
+   * BUG 1: Multiple singleton resources with different @singleton keys are not properly distinguished.
+   *
+   * This test demonstrates that resolveArmResources does not correctly return
+   * multiple singleton resources with different @singleton keys.
+   *
+   * Expected behavior: The API should return both Employee and CurrentEmployee
+   * as separate resources with their respective model types.
+   *
+   * Actual behavior: All resolved resources have type.name === "Employee",
+   * meaning CurrentEmployee is never properly identified.
+   *
+   * This test is marked with .skip because it demonstrates a bug in an external library
+   * that needs to be fixed before the test will pass.
+   */
+  it.skip("singleton resource - should return distinct resources for multiple singletons (BUG 1 - expected behavior)", async () => {
+    const program = await typeSpecCompile(multipleSingletonsSchema, runner);
+
+    // Use resolveArmResources API to get all resolved ARM resources
+    const provider = resolveArmResources(program);
+    const resolvedResources = provider.resources ?? [];
+
+    // Check that we have resources returned
+    ok(resolvedResources.length > 0, "Should have at least one resolved resource");
+
+    // Find resources by their model type name
+    const employeeResources = resolvedResources.filter(
+      (r) => r.type.name === "Employee"
+    );
+    const currentEmployeeResources = resolvedResources.filter(
+      (r) => r.type.name === "CurrentEmployee"
+    );
+
+    // BUG: This assertion fails because resolveArmResources returns all resources
+    // with type.name === "Employee" instead of distinguishing CurrentEmployee
+    ok(
+      currentEmployeeResources.length > 0,
+      "Should have at least one CurrentEmployee resource - BUG: resolveArmResources does not return CurrentEmployee"
+    );
+
+    // Verify that Employee resources are returned correctly
+    ok(
+      employeeResources.length > 0,
+      "Should have at least one Employee resource"
+    );
+
+    // Verify that the total count is correct (should be 2 distinct resources for CRUD operations)
+    // Note: The API may return multiple entries per resource for different operation paths
+    const uniqueResourceTypes = new Set(resolvedResources.map((r) => r.type.name));
+    strictEqual(
+      uniqueResourceTypes.size,
+      2,
+      "Should have exactly 2 unique resource types (Employee and CurrentEmployee)"
+    );
+  });
+
+  /**
+   * BUG 1: Multiple singleton resources with different @singleton keys are not properly distinguished.
```

Comment:
This is expected.  Singleton resources with different names are *not* different resource types.